src/github.sh

```diff
@@ -5,14 +5,16 @@ GITHUB_API_HEADER="Accept: application/vnd.github.v3+json"
 github::calculate_total_modifications() {
   local -r pr_number="${1}"
   local -r files_to_ignore="${2}"
+  local -r changed_file_weight="${3:-0}"
 
   if [ -z "$files_to_ignore" ]; then
     local -r body=$(curl -sSL -H "Authorization: token $GITHUB_TOKEN" -H "$GITHUB_API_HEADER" "$GITHUB_API_URL/repos/$GITHUB_REPOSITORY/pulls/$pr_number")
 
     local -r additions=$(echo "$body" | jq '.additions')
     local -r deletions=$(echo "$body" | jq '.deletions')
+    local -r changed_files=$(echo "$body" | jq '.changed_files')
 
-    echo $((additions + deletions))
+    echo $((additions + deletions + (changed_files * changed_file_weight)))
```

Comment:
To use an example, if there's a 10 file change and the `changed_file_weight` is set to 10, then we're adding an additional 100 lines to the calculated diff? Is that the intention?

If you're trying to calculate the complexity of the change, that might require understanding the number of line changes per total lines in all files. For example, a 100 line file having 30 lines changes would be complexity 30.

Reply:
Also, this change missing the `else` block here when `files_to_ignore` is set.